src/apis/gpt.py

```diff
@@ -96,29 +96,39 @@ def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
 
 
 class _GPTConversation:
-    def __init__(self, model: str, cost_callback, task_description, test_description, system_definition_examples: List[str] = ['executor', 'docarray', 'client']):
+    def __init__(self, model: str, cost_callback, messages: List[BaseMessage], print_stream, print_costs):
         self._chat = ChatOpenAI(
             model_name=model,
             streaming=True,
-            callback_manager=CallbackManager([AssistantStreamingStdOutCallbackHandler()]),
+            callback_manager=CallbackManager([AssistantStreamingStdOutCallbackHandler()] if print_stream else []),
             verbose=True,
             temperature=0,
         )
         self.cost_callback = cost_callback
-        self.messages: List[BaseMessage] = []
-        self.system_message = self._create_system_message(task_description, test_description, system_definition_examples)
-        if os.environ['VERBOSE'].lower() == 'true':
-            print_colored('system', self.system_message.content, 'magenta')
-
-    def chat(self, prompt: str):
-        chat_message = HumanMessage(content=prompt)
+        self.messages = messages
+        self.print_stream = print_stream
+        self.print_costs = print_costs
+        for message in messages:
+            if os.environ['VERBOSE'].lower() == 'true':
+                if isinstance(message, SystemMessage):
+                    print_colored('system - prompt', message.content, 'magenta')
+                elif isinstance(message, HumanMessage):
+                    print_colored('user - prompt', message.content, 'blue')
+                elif isinstance(message, AIMessage):
+                    print_colored('assistant - prompt', message.content, 'green')
```

Comment:
maybe `assistant - response` is better?

Reply:
I wanted to illustrate that this log output is part of the prompt we give to gpt.
Now we can differentiate between messages that are part of the prompt and generated messages.
It is new that we also have ai messages handed over as prompt

Reply:
Ah yes makes sense! Now I understand